### PR TITLE
Comment tags may break when switching Editor Mode

### PR DESCRIPTION
**Pull request by mattyrob**
## Description
See upstream ticket:
https://core.trac.wordpress.org/ticket/45947

Ticket details the issue and contains video showing the issue.

## Motivation and context
Backport of bug fix. To reproduce:

1. In current `develop` create a post with the content below in the Text tab of the Editor
2. Switch to the visual editor and back again.
3. Place the cursor within the `<!--more-->` tag and repeat step 2
4. Not how a <span> bookmark is now placed within the `<!--more-->` tag

```
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

<!--more-->

Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
```
Switch to the visual editor and back again.

## How has this been tested?
Upstream testing and commit 3 years ago, after applying the upstream patch this issue no longer occurs and the cursor placed within the `<!--more-->` tag is moved to the end of the tag on toggling between editors.

### Before
Screenshot 2022-05-01 at 11 47 30

### After
Screenshot 2022-05-01 at 11 47 04

## Types of changes
- Bug fix (possible)

### CODE REVIEW

File: src/wp-admin/js/editor.js [hunk 1]
```diff
@@ -382,11 +378,7 @@ window.wp = window.wp || {};
 		}
 
 		/**
-<<<<<<< HEAD:src/wp-admin/js/editor.js
-		 * @summary Get adjusted selection cursor positions according to HTML tags/shortcodes
-=======
-		 * Gets adjusted selection cursor positions according to HTML tags, comments, and shortcodes.
```
Comment (joyously) [inline src/wp-admin/js/editor.js]: Shouldn't you keep the extended explanation?

File: src/wp-admin/js/editor.js [hunk 2]
```diff
@@ -184,11 +184,7 @@ window.wp = window.wp || {};
 		}
 
 		/**
-<<<<<<< HEAD:src/wp-admin/js/editor.js
-		 * @summary Checks if a cursor is inside an HTML tag.
-=======
-		 * Checks if a cursor is inside an HTML tag or comment.
->>>>>>> b3a658edbc (TinyMCE: treat HTML comments same as tags when keeping cursor position. Fixes breaking the More and Nextpage tags when switching from Text to Visual.):src/js/_enqueues/wp/editor/base.js
```
Comment (joyously) [inline src/wp-admin/js/editor.js]: The extended explanation goes with this fix.

File: src/wp-admin/js/editor.js (hunk 1 above)
Comment (mattyrob) [reply src/wp-admin/js/editor.js]: You seem to be looking at the commit with the conflicts still in place, if you look at the final set of changes:
https://github.com/ClassicPress/ClassicPress/pull/990/files
You should see that the updated description has been retained.

Comment (joyously) [reply src/wp-admin/js/editor.js]: I click on *Files changed* and see only 1 file and 3 minor changes, none of which include that comment.
Comment tags may break when switching Editor Mode by mattyrob · Pull Request 990 · ClassicPressClassicPress-04

Comment (mattyrob) [reply src/wp-admin/js/editor.js]: Compare the file changes to the upstream Changeset:
https://core.trac.wordpress.org/changeset/44649

We would not usually add the commit messages into files if you are looking at lines like:
`b3a658edbc (TinyMCE: treat HTML comments same as tags when keeping cursor position. Fixes breaking the More and Nextpage tags when switching from Text to Visual.):src/js/_enqueues/wp/editor/base.js`

Comment (joyously) [reply src/wp-admin/js/editor.js]: Thanks for explaining what that is. Sorry for my confusion.

**Pull request merged**